src/connectionController.ts

```diff
@@ -425,6 +430,7 @@ export default class ConnectionController {
       );
 
     if (removeConfirmationResponse !== 'Confirm') {
+      await this.disconnect();
```

Comment:
If we don't disconnect here, does it have implications? In Compass we don't disconnect when a user declines re-authentication. I think it's fine either way, mostly curious on the reason for the change.

Reply:
So this is a result of my test failing [here](https://github.com/mongodb-js/vscode/pull/652/files#diff-507deef688826dd8859c67e31edf5e8e1ef8dcfa58ebd3718b3e8ffca85e63e8R397). I was expecting connectionController to be disconnected after a declined re-authentication. Later I checked this in the UI as well, without this the extension works as if we are still connected but then all the actions that you do like listing collections, etc just fails so it seemed like a reasonable thing to do. But I am open to doing it like compass as well, no strong preference here.